src/main/java/org/jboss/forge/arquillian/command/cube/CubeSetupCommand.java

```diff
@@ -0,0 +1,336 @@
+package org.jboss.forge.arquillian.command.cube;
+
+
+import org.jboss.forge.addon.facets.FacetFactory;
+import org.jboss.forge.addon.projects.ProjectFactory;
+import org.jboss.forge.addon.projects.facets.ResourcesFacet;
+import org.jboss.forge.addon.projects.ui.AbstractProjectCommand;
+import org.jboss.forge.addon.resource.FileOperations;
+import org.jboss.forge.addon.resource.ResourceFactory;
+import org.jboss.forge.addon.ui.command.UICommand;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.hints.InputType;
+import org.jboss.forge.addon.ui.input.UIInput;
+import org.jboss.forge.addon.ui.input.UISelectOne;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.metadata.WithAttributes;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.arquillian.api.ArquillianFacet;
+import org.jboss.forge.arquillian.api.TestFrameworkFacet;
+import org.jboss.forge.arquillian.api.YamlGenerator;
+import org.jboss.forge.arquillian.api.cube.CubeSetupFacet;
+import org.jboss.forge.arquillian.container.model.CubeConfiguration;
+import org.jboss.forge.furnace.util.Strings;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.jboss.forge.arquillian.util.StringUtil.getStringForCLIDisplay;
+
+public class CubeSetupCommand extends AbstractProjectCommand implements UICommand {
+
+    private CubeSetupFacet cubeSetupFacet;
+
+    @Inject
+    private ProjectFactory projectFactory;
+
+    @Inject
+    private FacetFactory facetFactory;
+
+    @Inject
+    private ResourceFactory resourceFactory;
+
+    @Inject
+    @WithAttributes(shortName = 't', label = "Type", type = InputType.DROPDOWN)
+    private UISelectOne<String> type;
```

Comment:
Type means openshift, docker, kubernetes, ... then since it is a known list I would suggest to just change to enum and provide autocompletition